### PR TITLE
Bug 2109673: Align Console plugin SDK package dependencies with generated code

### PR DESCRIPTION
**Pull request by vojtechszocs**
This PR aligns `dependencies` of the following Console plugin SDK packages with what's actually used within their generated `dist/<pkg>/lib` code (excluding type-only dependencies):

- `@openshift-console/dynamic-plugin-sdk` :arrow_right: add `classnames`, `immutable`, `reselect`, `lodash`
- `@openshift-console/dynamic-plugin-sdk-internal` :arrow_right: add `immutable`

This also excludes `@patternfly/react-icons` and `@patternfly/react-tokens` which are pulled in via `@patternfly/react-core`.

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/scripts/package-definitions.ts [hunk 1]
```diff
@@ -70,7 +70,12 @@ export const getCorePackage: GetPackageDefinition = (
     ...commonManifestFields,
     dependencies: {
       ...parseSharedModuleDeps(rootPackage, missingDepCallback),
-      ...parseDeps(rootPackage, ['typesafe-actions', 'whatwg-fetch'], missingDepCallback),
+      ...parseDeps(
+        rootPackage,
+        ['classnames', 'immutable', 'reselect', 'typesafe-actions', 'whatwg-fetch'],
```
Comment (spadgett) [inline frontend/packages/console-dynamic-plugin-sdk/scripts/package-definitions.ts:75]: @vojtechszocs I'm worried this list will get out of sync over time as the code changes. What happens to a plugin if a dependency is missing? Is there any way we can generate this list from the code rather than hard-coding?

Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/scripts/package-definitions.ts:75]: @spadgett correct - however I don't think detecting these dependencies is trivial. 

Chatted with @vojtechszocs and some plugins may already be depending on these missing dependencies (which is saving us). If one of these dependencies is missing and a plugin is using the utility that depends on that dep, they will get an error during their compilation. If a plugin is not using the utility that depends on one of these, they won't see any issues.

This change is our best effort to clean up the dependencies before 4.11 GA.

I think we have 2 options:
1. Use a bundler (rollup, webpack, etc)  which will hook into the processing of modules and analyze the import statements when generating the output
2. Use an external tool to scan the source files for import statements and generate the outputted package.json - this would require consideration for special cases like react-icons

@spadgett if you are ok with it, I'd like to follow up on this in another item. I'm leaning towards option 1 as rollup/webpack are well supported.

Comment (florkbr) [reply frontend/packages/console-dynamic-plugin-sdk/scripts/package-definitions.ts:75]: I've created a core ticket to track protecting the core SDK from missing dependencies: https://issues.redhat.com/browse/HAC-1468

We'll also probably want another ticket tracking pulling rollup into the console SDK - unless we want to make that effort part of pulling the entire core SDK back into the console sdk: https://issues.redhat.com/browse/HAC-557

Comment (vojtechszocs) [reply frontend/packages/console-dynamic-plugin-sdk/scripts/package-definitions.ts:75]: @spadgett I think most real world Console dynamic plugins use the template repo as the baseline which already contains most (if not all) of these dependencies in their `package.json` file.

> What happens to a plugin if a dependency is missing?

In that case, the webpack build of that plugin will most likely fail due to a missing module within the `node_modules` structure. But in order for that to happen, the given SDK code (export) must be referenced by the plugin (and therefore included in the webpack compilation) _and_ the required vendor dependency must be missing in the plugin's `node_modules` structure.

> Is there any way we can generate this list from the code rather than hard-coding?

Right now, we build Console dynamic plugin SDK packages (core, internal, webpack) by running `tsc` on different `tsconfig-{base,internal,webpack}.json` config files. This effectively mirrors the input source file structure (files included in the TS compilation) onto the generated output `dist` file structure.

We could use [TypeScript custom transformers](https://github.com/madou/typescript-transformer-handbook) e.g. via [`ttypescript`](https://github.com/cevek/ttypescript) but that sounds too complicated given our goal - all we need is a way to process the generated vendor module import statements and compare them with what's in the generated `package.json` file.

As @florkbr mentioned above, it's better to replace `tsc` with a module bundler like Rollup that is simple enough for our purposes (bundling a distributable library) while allowing us to write a simple Rollup plugin to process these generated vendor module import statements.

**TL;DR** - I think it's better to move from `tsc` to using a module bundler when building Console dynamic plugin SDK packages, which will then allow us to process generated code in a much simpler way (compared to trying to hook into TS compiler directly).

**Comment by upalatucci** [comment]
Hey guys!! 
Can we please speed up the review process a little bit for this pr :-) ? 
It's blocking [another pr](https://github.com/kubevirt-ui/kubevirt-plugin/pull/755). Tests are failing for dep issues and these changes will fix it.

**Review by spadgett — APPROVED** [review]
/lgtm
